dask/dataframe/rolling.py

```diff
@@ -91,31 +163,178 @@ def map_overlap(func, df, before, after, *args, **kwargs):
         ):
             raise ValueError("before and after must be positive integers")
 
-    if "token" in kwargs:
-        func_name = kwargs.pop("token")
-        token = tokenize(df, before, after, *args, **kwargs)
+    name = kwargs.pop("token", None)
+    parent_meta = kwargs.pop("parent_meta", None)
+
+    assert callable(func)
+    if name is not None:
+        token = tokenize(meta, before, after, *args, **kwargs)
     else:
-        func_name = "overlap-" + funcname(func)
-        token = tokenize(func, df, before, after, *args, **kwargs)
+        name = funcname(func)
```

Comment:
```suggestion
        name = "overlap-" + funcname(func)
```
Not sure if it matters, but this would be a bit more consistent with the origincal default name.

Reply:
Fixed in my last commit